### PR TITLE
Add parameters

### PR DESCRIPTION
**Pull request by anders-larsson**
This pull request adds a total of four new parameters to configure my_domain, smtpd_helo_required, smtpd_helo_restrictions and finally smtpd_recipient_restrictions.

With the default settings no additional entries are added to main.cf.

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -209,6 +213,7 @@
   if is_integer($main_mailbox_size_limit_real) == false { fail("main_mailbox_size_limit must be an integer and is set to <${main_mailbox_size_limit_real}>") }
   if $main_mailbox_size_limit_real + 0 < 0 { fail("main_mailbox_size_limit needs a minimum value of 0 and is set to <${main_mailbox_size_limit_real}>") }
   validate_string($main_mydestination_real)
+  if $main_mydomain != undef { if is_domain_name($main_mydomain) != true { fail("main_mydomain must be a domain name and is set to <${main_mydomain}>") } }
```
Comment (Phil-Friderici) [inline manifests/init.pp]: think using an  `and` between both conditions would ease the readability

**Comment by Phil-Friderici** [comment]
Thanks @anders-larsson ! All there, looks good to me. Please have a look at my comment regarding the validation.

**Comment by anders-larsson** [comment]
Cool. I'm not working tomorrow. Will fix on Monday. Thanks for fast review!

**Comment by Phil-Friderici** [comment]
@anders-larsson no worries, neither I am working ;)

**Comment by anders-larsson** [comment]
I've updated the first commit to include your requested changes.

File: spec/classes/init_spec.rb [hunk 2]
```diff
@@ -393,6 +399,54 @@
       }
     end
 
+    describe "with main_smtpd_helo_required set to 'yes' and main_smtpd_restrictions" do
```
Comment (Phil-Friderici) [inline spec/classes/init_spec.rb]: I see no mandatory connection of `$main_smtpd_helo_required` and `$main_smtpd_restrictions` in the code.
Please change the tests to become individual for each parameter too.

File: manifests/init.pp [hunk 3]
```diff
@@ -209,6 +210,7 @@
   if is_integer($main_mailbox_size_limit_real) == false { fail("main_mailbox_size_limit must be an integer and is set to <${main_mailbox_size_limit_real}>") }
   if $main_mailbox_size_limit_real + 0 < 0 { fail("main_mailbox_size_limit needs a minimum value of 0 and is set to <${main_mailbox_size_limit_real}>") }
   validate_string($main_mydestination_real)
+  if $main_mydomain != undef and is_domain_name($main_mydomain) != true { fail("main_mydomain must be a domain name and is set to <${main_mydomain}>") }
```
Comment (Phil-Friderici) [inline manifests/init.pp]: sorry for being pedantic, please use:

`if $main_mydomain != undef and is_domain_name($main_mydomain) == false { fail("main_mydomain must be a domain name and is set to <${main_mydomain}>") }`

**Comment by anders-larsson** [comment]
Updated! :)

**Comment by Phil-Friderici** [comment]
Thanks @anders-larsson ! Released in v1.4.0

**Pull request merged**